ReactWindows/ReactNative.Shared/UIManager/BorderedCanvasManager.cs

```diff
@@ -12,16 +13,58 @@ namespace ReactNative.UIManager
     /// View parent manager for bordered canvases.
     /// </summary>
     public abstract class BorderedCanvasManager<TCanvas> : BorderedViewParentManager<TCanvas>
-        where TCanvas : Canvas
+        where TCanvas : BorderedCanvas
     {
         /// <summary>
+        /// Checks if the Canvas has a Border already.
+        /// </summary>
+        protected override bool HasBorder(TCanvas view)
```

Comment:
>HasBorder [](start = 32, length = 9)

Maybe we should just leave HasBorder and GetOrCreateBorder abstract, then you can implement them on ReactViewManager. If we're going to keep this base class, I'd rather not tie it to a specific Canvas subclass.